### DISCUSSION TITLE
setting and setting  sys.path for custom directory imports

### DISCUSSION THREAD

**Discussion by dkebler** [berry-lang/berry#243] [General]
I'm trying to import some utility modules located in a different directory from the main be files

according to this
https://github.com/berry-lang/berry/wiki/Chapter-8#behavior-of-import

import will look for modules in `sys.path`

is `sys` a predined global?? 

because if I try to see or set the path using sys.path 

`print("system path ", str(sys.path))`

berry reports that 

`'sys' undeclared (first use in this function)`

Just unclear about the documenation and how to implment what says can be done.

**Comment by sfromis** [comment]
At least in a Tasmota context, you need to `import sys` to be able to use it.
```
import sys
print(sys.path())
```

**Reply by dkebler** [reply]
that function returns a list so 
```
> sys.path().push("/my/path")
> sys.path()
['/my/path']
```
but it doesn't survive running berry again (from the command line) so you'd have to reset it as part of a script say.  I bet there is a way to set the list at compile because  by default on my linux box it's   ['/usr/local/lib/berry/packages']

**Reply by s-hadinger** [reply]
I'm not sure there is an easy way.

Best option would be to extend the `berry.c` main source to support additional command line arguments, or get it from ENV variable.

**Comment by msx80** [comment]
I provided PR #289 for an additional command line argument since i also needed it.
Should be good for @dkebler too!
